rootston/output.c

```diff
@@ -37,9 +37,9 @@ static void rotate_child_position(double *sx, double *sy, double sw, double sh,
 }
 
 static void surface_for_each_surface(struct wlr_surface *surface, double lx,
-		double ly, float rotation, surface_iterator_func_t iterator,
+		double ly, float rotation, float alpha, surface_iterator_func_t iterator,
```

Comment:
Since #654 we have anoter user of the for_each functions, which currently does not require alpha.

I feel this should be in the user_data struct.

Reply:
I thought the same first but once I started to shift things around it looked more readable since we don't need to deref *data all the time.

Reply:
I can't see any point where `alpha` is used passed without also passing `user_data`.

And it is an unused argument to all iterator functions but `render_surface`.